strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-pypi-layout-provider/src/main/java/org/carlspring/strongbox/util/PypiMetadataFileParser.java

```diff
@@ -0,0 +1,89 @@
+package org.carlspring.strongbox.util;
+
+import com.google.common.collect.Maps;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.carlspring.strongbox.domain.PypiPackageInfo;
+import static org.carlspring.strongbox.domain.PypiPackageInfo.PypiMetadataKey;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.util.Map;
+import java.util.TreeMap;
+
+@Component
+public class PypiMetadataFileParser
```

Comment:
Could you please rename this to `PypiMetadataParser` and make this class work with `InputStream`, instead of files? We're actually not using `File`-s across our code base, (I think I should have mentioned this earlier, sorry!).